stdlib/public/core/FloatingPoint.swift.gyb

```diff
@@ -9,691 +9,840 @@
+  /// name "subnormal", and we follow that usage.
+  var isSubnormal: Bool { get }
+
+  /// True if and only if `self` is infinite.
+  ///
+  /// Note that `isFinite` and `isInfinite` do not form a dichotomy, because
+  /// they are not total.  If `x` is `NaN`, then both properties are `false`.
+  var isInfinite: Bool { get }
+
+  /// True if and only if `self` is NaN ("not a number"); this property is
+  /// true for both quiet and signaling NaNs.
+  var isNaN: Bool { get }
+
+  /// True if and only if `self` is a signaling NaN.
+  var isSignalingNaN: Bool { get }
+
+  /// The IEEE 754 "class" of this type.
+  var floatingPointClass: FloatingPointClassification { get }
+
+  /// True if and only if `self` is canonical.
+  ///
+  /// Every floating-point value of type Float or Double is canonical, but
+  /// non-canonical values of type Float80 exist, and non-canonical values
+  /// may exist for other types that conform to FloatingPoint.
+  ///
+  /// The non-canonical Float80 values are known as "pseudo-denormal",
+  /// "unnormal", "pseudo-infinity", and "pseudo-NaN".
+  /// (https://en.wikipedia.org/wiki/Extended_precision#x86_Extended_Precision_Format)
+  var isCanonical: Bool { get }
 }
 
-@_transparent
-@warn_unused_result
-public func != (lhs: ${Self}, rhs: ${Self}) -> Bool {
-  return Bool(Builtin.fcmp_une_FPIEEE${bits}(lhs._value, rhs._value))
+public enum FloatingPointSign: Int {
+  case plus
+  case minus
 }
 
-@_transparent
-@warn_unused_result
-public func <(lhs: ${Self}, rhs: ${Self}) -> Bool {
-  return Bool(Builtin.fcmp_olt_FPIEEE${bits}(lhs._value, rhs._value))
+public enum FloatingPointClassification {
+  case signalingNaN
+  case quietNaN
+  case negativeInfinity
+  case negativeNormal
+  case negativeSubnormal
+  case negativeZero
+  case positiveZero
+  case positiveSubnormal
+  case positiveNormal
+  case positiveInfinity
 }
 
-@_transparent
 @warn_unused_result
-public func > (lhs: ${Self}, rhs: ${Self}) -> Bool {
-  return Bool(Builtin.fcmp_ogt_FPIEEE${bits}(lhs._value, rhs._value))
+prefix func +<T: FloatingPoint>(x: T) -> T {
+  return x
 }
 
-@_transparent
 @warn_unused_result
-public func <= (lhs: ${Self}, rhs: ${Self}) -> Bool {
-  return Bool(Builtin.fcmp_ole_FPIEEE${bits}(lhs._value, rhs._value))
+prefix func -<T: FloatingPoint>(x: T) -> T {
+  return x.negated()
 }
 
-@_transparent
+%{
+binary_arithmetic = [
+('+', 'adding',      'add',      None),
+('-', 'subtracting', 'subtract', None),
+('*', 'multiplied',  'multiply', 'by'),
+('/', 'divided',     'divide',   'by')
+]
+}%
+
+%for op in binary_arithmetic:
 @warn_unused_result
-public func >= (lhs: ${Self}, rhs: ${Self}) -> Bool {
-  return Bool(Builtin.fcmp_oge_FPIEEE${bits}(lhs._value, rhs._value))
+func ${op[0]}<T: FloatingPoint>(lhs: T, rhs: T) -> T {
+  return lhs.${op[1]}(${op[3]+': ' if op[3] else ''}rhs)
 }
 
-@_transparent
-extension ${Self} : Comparable, Equatable {
+func ${op[0]}=<T: FloatingPoint>(lhs: inout T, rhs: T) {
```

Comment:
The standard library convention for generic constraints is to put a space on both sides of the colon, `T : FloatingPoint`.